### PR TITLE
[BUGFIX] fixing minor bug in sqlalchemy_execution_engine.py

### PR DESCRIPTION
**Pull request by SaeedFarahani**
Describe the bug
I found a minor bug while I was running our system integration tests and it happens when using some special rules so I think it's better to fix this issue in the main branch.

this is the expectation that GX raised:

great_expectations/execution_engine/execution_engine.py", line 550, in _process_direct_and_bundled_metric_computation_configurations\n raise gx_exceptions.MetricResolutionError(\ngreat_expectations.exceptions.exceptions.MetricResolutionError: 'Subquery' object has no attribute 'subquery'\n", "exception_message": "'Subquery' object has no attribute 'subquery'",

Environment (please complete the following information):

Operating System: MacOS
Great Expectations Version: 0.17.5
Datasource: SQL

### CODE REVIEW

File: great_expectations/compatibility/sqlalchemy.py [hunk 1]
```diff
@@ -9,8 +9,10 @@
 
 try:
     import sqlalchemy
+    from sqlalchemy.sql.selectable import Subquery
```
Comment (alexsherstinsky) [inline great_expectations/compatibility/sqlalchemy.py]: Hello, @SaeedFarahani -- thank you for this contribution.  One small change request: Please create a separate section for the new `Subquery` attribute.  Thank you.

Comment (SaeedFarahani) [reply great_expectations/compatibility/sqlalchemy.py]: Hi @alexsherstinsky 
It's my pleasure, I fixed the sections

Comment (alexsherstinsky) [reply great_expectations/compatibility/sqlalchemy.py]: @SaeedFarahani Much appreciated!  The code looks good -- once all the tests pass, I will approve.  Thank you very much!

File: great_expectations/compatibility/sqlalchemy.py [hunk 2]
```diff
@@ -12,6 +12,11 @@
 except ImportError:
     sqlalchemy = SQLALCHEMY_NOT_IMPORTED
 
+try:
+    from sqlalchemy.sql.selectable import Subquery
+except ImportError:
```
Comment (alexsherstinsky) [inline great_expectations/compatibility/sqlalchemy.py]: @SaeedFarahani We also need to add `AttributeError` here -- please see other examples that use it in this module.  Thanks!

Comment (SaeedFarahani) [reply great_expectations/compatibility/sqlalchemy.py]: @alexsherstinsky  I fixed this as well,
Thank you in advance

**Comment by alexsherstinsky** [comment]
Everything is great; just waiting for the release to finish; will unblock thereafter. Thank you.

**Review by alexsherstinsky — APPROVED** [review]

**Comment by SaeedFarahani** [comment]
@alexsherstinsky  I rebased the branch, could you merge it, please?

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 3]
```diff
@@ -722,7 +723,9 @@ def get_domain_records(  # noqa: C901, PLR0912, PLR0915
 
             # SQLAlchemy 2.0 deprecated select_from() from a non-Table asset without a subquery.
             # Implicit coercion of SELECT and textual SELECT constructs into FROM clauses is deprecated.
-            if not isinstance(selectable, sa.Table):
+            if not isinstance(selectable, sa.Table) and not isinstance(
+                selectable, Subquery
+            ):
```
Comment (Kilo59) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]:
```suggestion
            if not isinstance(selectable, (sa.Table, Subquery)):
```
I think we can combine these `isinstance()` checks.

**Pull request merged**